solr/core/src/java/org/apache/solr/store/shared/metadata/SharedShardMetadataController.java

```diff
@@ -146,20 +148,18 @@ public void cleanUpMetadataNodes(String collectionName, String shardName) {
     }
   }
 
-  private void createPersistentNodeIfNonExistent(String path, byte[] data) {
+  private void createPersistentNode(String path, byte[] data) {
     try {
       if (!stateManager.hasData(path)) {
-        try {
-          stateManager.makePath(path, data, CreateMode.PERSISTENT, /* failOnExists */ false);
-        } catch (AlreadyExistsException e) {
-          // it's okay if another beats us creating the node
-        }
+        stateManager.makePath(path, data, CreateMode.PERSISTENT, /* failOnExists */ true);
```

Comment:
#makePath will also throw an AlreadyExistsException if the node exists and failOnExists=true